pinot-common/src/main/java/com/linkedin/pinot/pql/parsers/pql2/ast/IdentifierAstNode.java

```diff
@@ -19,16 +19,29 @@
  * AST node for identifiers (column names).
  */
 public class IdentifierAstNode extends BaseAstNode {
+  private String _expression;
   private String _name;
 
-  public IdentifierAstNode(String name) {
-    _name = name;
+  public IdentifierAstNode(String expression) {
+    _expression = expression;
+    if (
+      expression.charAt(0) == '`'
+      && expression.charAt(expression.length()-1) == '`'
+    ) {
+      _name = expression.substring(1, expression.length()-1);
+    } else {
+      _name = expression;
+    }
   }
 
   public String getName() {
     return _name;
   }
 
+  public String getExpression() {
```

Comment:
Is there a particular reason why we're introducing the concept of an expression as opposed to just using the name? In which scenarios would we need the quoted identifier with backticks?

Reply:
Great question, sorry I should have called that out.  The reason is that the expression is parsed and re-serialized back into a string.  Currently this happens here: https://github.com/linkedin/pinot/blob/a4b05037253ca5d8c1bfa7b21b767deaca6d105c/pinot-common/src/main/java/com/linkedin/pinot/pql/parsers/pql2/ast/GroupByAstNode.java#L40-L47

If the ability is needed to go from expression tree back into string form, we would need to be able to distinguish between the name and expression of the identifier.

Without getExpression() this:
```
select count(*) from table group by `this.is.my.identifier`
```

would be parsed and toStringed with an invalid identifier:
```
select count(*) from table group by this.is.my.identifier
```

Reply:
Ah good point, completely forgot about server-side reparse for expressions.